include/knowhere/comp/thread_pool.h

```diff
@@ -48,23 +51,33 @@ class ThreadPool {
     };
 
  public:
-    explicit ThreadPool(uint32_t num_threads, const std::string& thread_name_prefix)
-        : pool_(folly::CPUThreadPoolExecutor(
-              num_threads,
-              std::make_unique<
-                  folly::LifoSemMPMCQueue<folly::CPUThreadPoolExecutor::CPUTask, folly::QueueBehaviorIfFull::BLOCK>>(
-                  num_threads * kTaskQueueFactor),
-              std::make_shared<LowPriorityThreadFactory>(thread_name_prefix))) {
+    explicit ThreadPool(uint32_t num_threads, const std::string& thread_name_prefix, QueueType queueT = QueueType::LIFO)
```

Comment:
Personally, I would introduce `static ThreadPool CreateFIFO()` and `static ThreadPool CreateLIFO()`. These are two very different behaviors which one does not control via a boolean flag, and it make the code more readable.

Reply:
Sounds good